pkg/buildpack/buildpack.go

```diff
@@ -17,73 +17,110 @@ import (
 	"github.com/buildpacks/pack/pkg/dist"
 )
 
-type Blob interface {
-	// Open returns a io.ReadCloser for the contents of the Blob in tar format.
-	Open() (io.ReadCloser, error)
-}
-
 //go:generate mockgen -package testmocks -destination ../testmocks/mock_buildpack.go github.com/buildpacks/pack/pkg/buildpack Buildpack
 
-type Buildpack interface {
+type Buildpack interface { // TODO: this should ideally have a more generic name since it could be a buildpack OR an extension
 	// Open returns a reader to a tar with contents structured as per the distribution spec
-	// (currently '/cnbs/buildpacks/{ID}/{version}/*', all entries with a zeroed-out
+	// (currently '/cnb/buildpacks/{ID}/{version}/*', all entries with a zeroed-out
 	// timestamp and root UID/GID).
 	Open() (io.ReadCloser, error)
-	Descriptor() dist.BuildpackDescriptor
+	Descriptor() Descriptor
 }
 
-type buildpack struct {
-	descriptor dist.BuildpackDescriptor
+type Descriptor interface {
+	EnsureStackSupport(stackID string, providedMixins []string, validateRunStageMixins bool) error
+	EscapedID() string
+	Kind() string
+	ModuleAPI() *api.Version
+	ModuleInfo() dist.BuildpackInfo
+	ModuleOrder() dist.Order
+	ModuleStacks() []dist.Stack
```

Comment:
Are we able to remove the `Module` prefix to the getter methods? They fill a tad awkward.

Reply:
Updated in bd37213a